packages/eslint-plugin-template/src/rules/no-negated-async.ts

```diff
@@ -27,54 +27,49 @@ export default createESLintRule<Options, MessageIds>({
   },
   defaultOptions: [],
   create(context) {
-    const parserServices = getTemplateParserServices(context);
+    ensureTemplateParser(context);
     const sourceCode = context.getSourceCode();
 
-    return parserServices.defineTemplateBodyVisitor({
-      ['BindingPipe[name=async]'](node: any) {
-        if (node.parent.type === 'PrefixNot') {
-          const additionalOffset =
-            node.parent.parent.type === 'Interpolation' ? -1 : 0;
+    return {
+      'PrefixNot > BindingPipe[name=async]'({
```

Comment:
Just noticed that I missed quotes in `name=async`, but I believe that isn't it a problem, right @JamesHenry?